time/src/main/java/net/jqwik/time/internal/properties/arbitraries/valueRanges/ValueRange.java

```diff
@@ -0,0 +1,61 @@
+package net.jqwik.time.internal.properties.arbitraries.valueRanges;
+
+public abstract class ValueRange<T> {
+
+	private T min;
+	private T max;
+
+	protected void exceptionCheck(Parameter parameter) {
+		//Override if needed
+		//do nothing in default case
+	}
+
+	protected void minMaxChanger(Parameter parameter) {
```

Comment:
There's only one subclass that does not swap min and max. So swapping could be the default implementation. Moreover, the name could be more telling, e.g. "ensureMinIsBeforeMax" or something similar.